substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -1843,14 +1875,44 @@ impl<T: Config> Pallet<T> {
 		Self::check_exposures()?;
 		Self::check_paged_exposures()?;
 		Self::check_ledgers()?;
+		Self::check_locks()?;
 		Self::check_count()
 	}
 
+	/// Invariants:
+	///
+	/// * Staking locked funds for every bonded stash should be the same as its ledger's total.
+	///
+	/// NOTE: some of these checks result in warnings only. Once
+	/// <https://github.com/paritytech/polkadot-sdk/issues/3245> is resolved, turn warns into check
+	/// failures.
+	fn check_locks() -> Result<(), TryRuntimeError> {
```

Comment:
Does this pass now in all networks, excluding the few corrupted ledgers? it should

Reply:
On my list for a quick check tomorrow.